### PR TITLE
Verify Email landing, confirmation, and error pages

### PR DESCRIPTION
**Pull request by kiwansim**
* render different screens depending on the case returned by auth server's verifyEmail response 
* add Expired Link page, Successful Verification & Redirect page, Re-request Verification Email page 
* fix cors errors
* **does not** handle very specific edge cases -- sc stories forthcoming
    * user is not signed into WOW, clicks on a verification email (expired or valid link), user object from auth server does not get returned and the user's email doesn't display on landing page
    * user is signed into WOW with kiwan@justfix.org. user clicks on verification email for kiwan+001@justfix.org. auth server returns user object associated with logged in account. 

`shakao/auth-endpoints` will remain as the feature branch for now since demo site & auth server point to it. 

### User clicks expired link
https://github.com/JustFixNYC/who-owns-what/assets/17574626/01a918ad-2431-4fa8-ad8f-d967f214a79c

### User clicks valid link
https://github.com/JustFixNYC/who-owns-what/assets/17574626/79f1a9e0-0ab7-452d-8c06-0870ca0e809f

[sc-12484] [sc-12719] [sc-12158] [sc-12403]

### CODE REVIEW

File: client/src/containers/VerifyEmailPage.tsx [hunk 1]
```diff
@@ -1,44 +1,148 @@
...
+    const delayInterval = delaySeconds * 100;
+
+    setInterval(() => {
+      timeLeft--;
+      document.getElementById("countdown")!.textContent = timeLeft.toString();
+      if (timeLeft <= 0) {
+        document.location.href = redirectUrl;
+      }
+    }, delayInterval);
+  };
```
Comment (austensen) [inline client/src/containers/VerifyEmailPage.tsx:48]: when testing this out it went down to -1 on the page before redirecting

File: client/src/containers/VerifyEmailPage.tsx [hunk 2]
```diff
@@ -1,44 +1,148 @@
...
+
+  const renderConfirmationPage = () => {
+    const resendEmailPage = (
+      <Fragment>
+        <Trans render="h3" className="text-center">
+          Check your email inbox & spam
+        </Trans>
+        <br />
+        <Trans className="text-center">
+          Click the link we sent to verify your email address {!!user && user.email}. It may take a
```
Comment (austensen) [inline client/src/containers/VerifyEmailPage.tsx:111]: I think this is just a lingui thing, but in testing this I got `{0}` for the email
image

**Review by austensen — APPROVED** [review]
This all looks great! tested it out on the pr deploy link and everything worked well. just a couple possible minor things in comments

File: client/src/containers/VerifyEmailPage.tsx [hunk 3]
```diff
@@ -1,44 +1,148 @@
...
+      <Fragment>
+        <Trans className="text-center" render="h3">
+          Your email is now verified
+        </Trans>
+        <br />
+        <div className="text-center">
+          <Trans className="text-center">You will be redirected back to Who Owns What in:</Trans>
+          <br>{updateCountdown()}</br>
+          <Trans className="d-flex justify-content-center">
+            <span id="countdown">{delaySeconds}</span> seconds
```
[the remark this answers is not in the document]
Comment (kiwansim) [reply client/src/containers/VerifyEmailPage.tsx:128]: i'm going to leave it as `<span>` so the `document.getElementById` is something I can do to dynamically update the counts. Plural's value only takes number or string

**Pull request merged**